feat(indexing): simplify and rely on contentDigest (#162)
algolia/gatsby-plugin-algolia: CHANGELOG.md

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

For migrating to v1 version smoothly:

1. add `id` instead of `objectID` in your query
2. query for `internal { contentDigest }` and do not transform it
3. set `enablePartialUpdates` to `true`
4. set `matchFields` to `['internal.contentDigest']`

> Once these changes are made, you can index once before updating. Then once updated, you remove the `enablePartialUpdates` and `matchFields` updates from the config.

Reviewer: ```suggestion
You can now index once before updating, then remove the `enablePartialUpdates` and `matchFields` updates from the config.
```
